Assets/Scripts/Player/PlayerMovement.cs

```diff
@@ -0,0 +1,30 @@
+using UnityEngine;
+
+public sealed class PlayerMovement : MonoBehaviour
+{
+    [SerializeField] private CharacterController characterController;
+    [SerializeField] private float movingSpeed;
+
+
+    private Vector2 _moveDirection;
+    private float _horizontal;
+    private float _vertical;
+
+    private void Update()
+    {
+        SetInput();
+        SetMoveDirection();
+    }
+
+    private void SetInput()
+    {
+        _horizontal = Input.GetAxisRaw("Horizontal");
+        _vertical = Input.GetAxisRaw("Vertical");
+    }
+
+    private void SetMoveDirection()
+    {
+        _moveDirection = (transform.right * _horizontal + transform.up * _vertical) * movingSpeed;
+        characterController.Move(_moveDirection * Time.deltaTime);
+    }

```

Comment:
If you really want to avoid using variables, you could *even* rewrite the SetInput method to have a vector2 return type.

```suggestion
    private Vector2 GetInput()
    {
        var input2D = Vector2.zero;
        input2D.x = Input.GetAxisRaw("Horizontal");
        input2D.y = Input.GetAxisRaw("Vertical");
        return input2D;
    }

    private void SetMoveDirection()
    {
        _moveDirection = (transform.right * GetInput().x + transform.up * GetInput().y) * movingSpeed;
        characterController.Move(_moveDirection * Time.deltaTime);
    }
```
If you don't like the method being used instantly in SetMoveDirection for the x and y, you can also use an `out Vector2 input2D` in GetInput and then when calling it in SetMoveDirection you can use the out variable that you create there for the x and y! If you want an example of that as well let me know.